### PR TITLE
Try to find a comment to edit before creating

### PR DESCRIPTION
**Pull request by refack**
* Add `@octokit/graphql`

### CODE REVIEW

File: lib/github-comment.js [hunk 1]
```diff
@@ -1,16 +1,55 @@
 'use strict'
 
 const githubClient = require('./github-client')
+const GQL = require('./github-graphql-client')
+
+const getPRComments = `query getPRComments($owner: String!, $repo: String!, $number: Int!, $cursor: String){
+  repository(owner: $owner, name:  $repo) {
+    pullRequest(number: $number) {
+      comments(first: 20, after:$cursor) {
```
Comment (refack) [inline lib/github-comment.js]:
```suggestion
      comments(first: 100, after:$cursor) {
```
I just put 20 to see if the recursive querying works.

File: lib/github-comment.js [hunk 2]
```diff
@@ -1,16 +1,55 @@
...
-  }, (err) => {
-    if (err) {
-      logger.error(err, 'Error while creating comment on GitHub')
+  exports.getFirstBotComment({ owner, repo, number, logger }).then((comment) => {
+    if (comment) {
+      const { id, body: oldBody } = comment
+      const newBody = `${oldBody}\n${body}`
+      return githubClient.issues.editComment({ owner, repo, id, body: newBody })
     }
+    return githubClient.issues.createComment({ owner, repo, number, body })
```
Comment (phillipj) [inline lib/github-comment.js]: For sure like the looks of using GraphQL and promises 👍

File: lib/github-comment.js [hunk 3]
```diff
@@ -1,16 +1,55 @@
...
+        pageInfo {
+          endCursor
+          hasNextPage
+        }
+      }
+      labels(first: 15) {
+        nodes {
+          name
+        }
+      }
```
Comment (phillipj) [inline lib/github-comment.js]: Planning to use these labels for something?

**Comment by phillipj** [comment]
Leaning towards keeping these changes instead of what I suggested in https://github.com/nodejs/github-bot/pull/226, especially when it comes to talking to the GitHub API.

The other difference from the mentioned PR, is tests. You got any plans for more tests?

**Comment by refack** [comment]
> You got any plans for more tests?

That's my current WIP... mocking the GQL calls. (H/T to @gr2m for https://github.com/octokit/graphql.js/#writing-tests)

File: lib/github-graphql-client.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+'use strict'
+
+const GitHubGQL = require('@octokit/graphql').defaults({
+  headers: {
+    'user-agent': 'Node.js GitHub Bot v1.0-beta',
+    authorization: 'token ' + process.env.GITHUB_TOKEN || 'invalid-placeholder-token'
```
Comment (gr2m) [inline lib/github-graphql-client.js]:
```suggestion
    authorization: 'token ' + (process.env.GITHUB_TOKEN || 'invalid-placeholder-token')
```

```
> 'token ' + undefined || 'invalid-placeholder-token'
< token undefined
> 'token ' + (undefined || 'invalid-placeholder-token')
< token invalid-placeholder-token
```

Comment (refack) [reply lib/github-graphql-client.js]: Ack!

File: lib/github-comment.js (hunk 3 above)
Comment (refack) [reply lib/github-comment.js]: If we want to recognize a `bot-out` label, for complete opt-out.

**Comment by gr2m** [comment]
Holler if you run into any blockers or have any questions about GitHub APIs / Octokit, happy to help

**Comment by phillipj** [comment]
Any updates on this? I fully understand if you've got other things that needs attention -- if so, maybe we should consider merging https://github.com/nodejs/github-bot/pull/226 at first to reduce the amount of CI comment noise that @cjihrig was concerned about? 🤷‍♂️ ..then merge this as an improvement as soon as it's ready.

**Comment by phillipj** [comment]
@refack would you mind if I dived into getting a test in place, with the goal of getting this merged & deployed?

**Comment by phillipj** [comment]
@refack as I would _really_ appreciate getting this landed soon, I took the freedom of rebasing with master to fix a `package-lock.json` merge conflict and pushed a couple of extra commits with missing GraphQL fixtures & tests -- sorry in advance if I crossed a line by doing that.

PTAL

**Comment by phillipj** [comment]
@refack @cjihrig is this still something we want rolled out or are you happy with how things work currently?

**Comment by cjihrig** [comment]
I'd still prefer if the bot limited its commenting when possible.

**Comment by phillipj** [comment]
Alrighty then, let's give this a test round to see how it behaves 🚀

Rebased and force push to fix conflict in `package-lock.json`.

**Comment by phillipj** [comment]
This is now live in production.

**Pull request merged**